ssh/tailssh/incubator.go

```diff
@@ -121,22 +122,7 @@ func (ss *sshSession) newIncubatorCommand() (cmd *exec.Cmd) {
 		if isShell {
 			incubatorArgs = append(incubatorArgs, "--shell")
 		}
-		// Only the macOS version of the login command supports executing a
```

Comment:
I don't see any benefit to determining the login command here and then passing it to the incubator as a flag.

Reply:
the intent was being able to write tests, i'd recommend leaving it as is so that we can actually write those tests.

Reply:
It feels awkward to complicate the code for a hypothetical test, whose shape isn't clear to me...

Reply:
Also, if the intent is to move all the platform-specific bits out of the `beIncubator()` function, then we should really have done the `loginArgs()` logic inside `newIncubatorCommand()` too.

Reply:
And of course, if the intent is to test just the `beIncubator()` function, the more stuff we move out of `beIncubator()`, the less we end up testing :)

Reply:
Put differently ... this does seem worth testing, but what feels most useful is an integration test on the actual platforms that matter. I wonder if it's worth building docker images from the relevant distributions and testing in those?

Reply:
I added an integration test. You can see the results in CI [here](https://github.com/tailscale/tailscale/actions/runs/8856869266/job/24323661368?pr=11868).